compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -698,6 +698,9 @@ static void addLowerToLLVMPasses(OpPassManager &passManager,
 
   if (enableAArch64SME) {
     // (Arith, Vector) -> ArmSME
+    passManager.addPass(mlir::arm_sme::createVectorLegalizationPass());
+    passManager.addNestedPass<func::FuncOp>(createCanonicalizerPass());
+    passManager.addNestedPass<func::FuncOp>(createCSEPass());
```

Comment:
Do we run cleanup passes at the end of the if-statement?

Reply:
There's more cleanups that apply before things are lowered to ArmSME and SCF, so it's best to apply these right after the legalizer runs (which is still working on vector + arith ops).

Reply:
I see, thanks